### PR TITLE
[RN] Update Gradle

### PR DESCRIPTION
**Pull request by zbettenbuk**
This PR updates gradle to the latest versions (4.4, plugin version 3.1.2). This is required for the Google sign-in lib we want to use for streaming.

Some notes:
- The new plugin version doesn't seem to have the tasks process${buildTypeHere}Resources (any anything similar) task, so font icons and sounds were missing from the build with the new versions. To come around this I got gradle to copy over the assets into the assets folder in src so then they get included with the standard build process.
- With that said, since the assets folder is generated I added that to gitignore
- Everything seems to be working this way, but I'm not sure how to test the ```bundleJsAndAssetsTaskName``` and ```currentBundleTask``` tasks, where I applied a similar logic, but got one phase (task) later. Any input on this is welcome!

If anyone has a better task to execute the font and sound copy tasks on in mind, don't keep the info!

### CODE REVIEW

**Comment by zbettenbuk** [comment]
[jitsi-gradle-tasks.txt](https://github.com/jitsi/jitsi-meet/files/2159463/jitsi-gradle-tasks.txt)

Attached the list of available gradle tasks for a basis of discussion

File: android/build.gradle [hunk 1]
```diff
@@ -144,7 +144,6 @@ allprojects {
 }
 
 ext {
-    buildToolsVersion = "26.0.2"
```
Comment (saghul) [inline android/build.gradle]: Is this change intentional?

Comment (zbettenbuk) [reply android/build.gradle]: Yes, the new version doesn't care about it and uses its default one

File: android/sdk/build.gradle [hunk 2]
```diff
@@ -52,24 +52,20 @@ gradle.projectsEvaluated {
...
             from("${projectDir}/../../fonts/jitsi.ttf")
             from("${projectDir}/../../node_modules/react-native-vector-icons/Fonts/")
-            into("${bundlePath}/assets/fonts")
+            into("${projectDir}/src/main/assets/fonts")
         }
 
-        currentFontTask.dependsOn("merge${buildNameCapitalized}Resources")
-        currentFontTask.dependsOn("merge${buildNameCapitalized}Assets")
+        runBefore("generate${buildNameCapitalized}Resources", currentFontTask)
+        runBefore("generate${buildNameCapitalized}Assets", currentFontTask)
```
Comment (lyubomir) [inline android/sdk/build.gradle]: Assets are easy without copying to android/sdk/src/main/assets and .gitignore-ing then:
1) Change `android.buildTypes.all { buildType ->` above to `android.libraryVariants.all { variant ->`
2) Add `def buildType = variant.buildType`
3) Change `into("${projectDir}/src/main/assets/fonts")` to `into("${variant.mergeAssets.outputDir}/fonts")`
4) Change `currentFontTask.dependsOn("merge${buildNameCapitalized}Assets")` to `currentFontTask.dependsOn(variant.mergeAssets)`. I'm not exactly sure why you deleted the mergeAssets dependency. The mergeResources dependencies doesn't make sense to me because the currentFontTask produces assets only, not resources.
5) Change `runBefore("generate${buildNameCapitalized}Assets", currentFontTask)` to `variant.outputs.each { output -> output.processResources.dependsOn(currentFontTask) }`

Which eventually results in pretty much the old source code without figuring out what `process${buildNameCapitalized}Resources` is called now.

File: android/sdk/build.gradle [hunk 3]
```diff
@@ -137,10 +129,8 @@ gradle.projectsEvaluated {
         currentBundleTask.dependsOn("merge${buildNameCapitalized}Resources")
         currentBundleTask.dependsOn("merge${buildNameCapitalized}Assets")
 
-        runBefore("processArmeabi-v7a${buildNameCapitalized}Resources", currentBundleTask)
-        runBefore("processX86${buildNameCapitalized}Resources", currentBundleTask)
-        runBefore("processUniversal${buildNameCapitalized}Resources", currentBundleTask)
-        runBefore("process${buildNameCapitalized}Resources", currentBundleTask)
+        runBefore("package${buildNameCapitalized}Resources", currentBundleTask)
```
Comment (lyubomir) [inline android/sdk/build.gradle]: The tricky part is currentBundleTask because it produces both assets and resources. Again, it can be rewritten with `variant` as above without figuring out the new name for `process${buildNameCapitalized}Resources` because `output.processResources` knows it.

File: android/sdk/build.gradle (hunk 2 above)
Comment (zbettenbuk) [reply android/sdk/build.gradle]: Wow, didn't know you know Gradle this much... should have just handed over the task to you ;) Thanks, this works perfectly!

Comment (zbettenbuk) [reply android/sdk/build.gradle]: Wow, didn't know you know Gradle this much... should have just handed over the task to you ;) Thanks, this works perfectly!

**Comment by zbettenbuk** [comment]
@lyubomir updated the PR but mainly for testing, not for review yet

File: android/build.gradle (hunk 1 above)
Comment (saghul) [reply android/build.gradle]: Let's be specific. For reproducible builds. We have had crashes in the past due to the version here.

Comment (saghul) [reply android/build.gradle]: Doh, ignore this, GH wasn't showing me the next line.

**Comment by zbettenbuk** [comment]
Thanks @saghul, you got the essence of this change (other than the obvious version updates).

@lyubomir: Tested the change, works fine and it can now create a release nicely.

So the essence of this change is that we don't tap into the build process and try to put files in the right place at the right time during build, hoping that they'll be there by the time the process pick them up, but we prepare everything for an uninterrupted build before the build even starts, and then let it run natively without interfering with the build and the intermediates directories (which in my world are something not for tinkering with).

This way we can likely avoid our build process to brake again when the build workflow changes due to a future version update or similar. Backside though is the set of gitignored files that we need to handle.

Now regarding your question 3, I have no idea what to answer for that yet. After a quick smoke test everything seems to work fine now, we have calendar access (react-native-xxx lib), we have fonts (react-native-xxx lib) and sounds too (also a react-native-xxx lib). Maybe we can say we cross that bridge when we get to it?
